File: LeetCode/485-max-consecutive-ones/ex_485.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max_count = 0;
        int count = 0;
        for (auto & n : nums) {
            if (n == 1) {
                ++count;
                max_count = max(max_count, count);
            }
            else    count = 0;
        }
        return max_count;
    }
};

File: LeetCode/485-max-consecutive-ones/ex_485.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_count = count = 0
        for n in nums:
            if n == 1:
                count += 1
                max_count = max(max_count, count)
            else:
                count = 0
        return max_count